### PR TITLE
refactor(migrates): add some abstract methods for migrates

### PR DESCRIPTION
**Pull request by yhilmare**
#### What type of PR is this?

type-refactor

#### What this PR does / why we need it:

as the increment of the team organization and individual organization, the time cost of resource migration is more and more unacceptable, so that we should migrate all resources when user login or organization created.

We refactor the migrates framework to archive the goal mentioned, so that we can reuse the ability of migrate framework when migrate organization or user's data.

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/BootstrapSchemaHistoryRepository.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * schema history persistent, the table schema define in migrate_schema_history_table_template.sql
+ * 
+ * @author yizhou.xw
+ * @version : SchemaHistoryRepository.java, v 0.1 2021-03-26 14:26
+ */
+@Slf4j
+public class BootstrapSchemaHistoryRepository implements SchemaHistoryRepository {
```
Comment (yizhouxw) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/BootstrapSchemaHistoryRepository.java:51]: how about JdbcSchemaHistoryRepository ?

Comment (yhilmare) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/BootstrapSchemaHistoryRepository.java:51]: which is based on scenario, `BootstrapSchemaHistoryRepository` is used on `migrate_schema_history`, the other impl of `SchemaHistoryRepository` can also be based on jdbc but different table.

Comment (MarkPotato777) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/BootstrapSchemaHistoryRepository.java:51]: LGTM

**Review by MarkPotato777 — DISMISSED** [review]
LGTM

**Review by MarkPotato777 — APPROVED** [review]

**Pull request merged**